frontend/src/metabase/dashboard/components/DashCard/DashCardActionButtons/ChartSettingsButton.tsx

```diff
@@ -32,6 +32,7 @@ function ChartSettingsButton({
       tall
       triggerElement={
         <DashCardActionButton
+          as="span"
```

Comment:
Can we make it an actual `button`? Or at least set `role="button"` on it?

Reply:
`as="button"` seems to change the cursor from pointer to the default 🤔 

I'm actually wondering if we really want it to be a button, the html would be like this:
(`a > button`)
<img width="740" alt="image" src="https://github.com/metabase/metabase/assets/1914270/8ebd22ff-2c75-41eb-ba97-454163b0672b">

The issue is that `triggerElement` is wrapped in a `Trigger` component which is a `a`: https://github.com/metabase/metabase/blob/f3bc571fc80381d773d8ffda101cf2547c8c2cdb/frontend/src/metabase/components/Triggerable/Triggerable.jsx#L10
Causing the nested `a`.

A better solution would probably be adding a prop to `ModalWithTrigger` such as `triggerAs` but it seems to be a deprecated component so probably not worth changing its behaviour.
If we plan on replacing its use here with something else then maybe we can live with the warning until then?